### PR TITLE
get admob node from main_scene

### PR DESCRIPTION
**Pull request by syntaxerror247**
This PR improves the logic behind getting admob node.
It enables getting admob node from main scene instead of currently opened scene and fixes issue mentioned in #17

### CODE REVIEW

File: admob/addon_template/AdmobExportPlugin.gd [hunk 1]
```diff
@@ -59,7 +59,8 @@ class AndroidExportPlugin extends EditorExportPlugin:
 
 
 	func _get_android_manifest_application_element_contents(platform: EditorExportPlatform, debug: bool) -> String:
-		var __admob_node: Admob = _get_admob_node(EditorInterface.get_edited_scene_root())
+		var main_scene = load(ProjectSettings.get_setting("application/run/main_scene")).instantiate()
```
Comment (cengiz-pz) [inline admob/addon_template/AdmobExportPlugin.gd]: @syntaxerror247, what do you think about checking the main scene if there is no Admob node in the "edited scene"?  That way we would (1) be "backward compatible" in terms of habits of existing user base & (2) search both the edited and main scenes before throwing an error.

**Review by cengiz-pz — COMMENTED** [review]
Great idea!  Let me know what you think about my suggestion.  Thanks.

File: admob/addon_template/AdmobExportPlugin.gd (hunk 1 above)
Comment (syntaxerror247) [reply admob/addon_template/AdmobExportPlugin.gd]: Yes I think it would be good

Comment (syntaxerror247) [reply admob/addon_template/AdmobExportPlugin.gd]: @cengiz-pz I am going to move loading of main_scene inside condition block before checking in main_scene. Okay? To save some negligible processing power 😅

Comment (cengiz-pz) [reply admob/addon_template/AdmobExportPlugin.gd]: Yes, that would be great.  And printing an error message if node is not found on either scene would be a great way to let the users know that something failed.

**Review by cengiz-pz — APPROVED** [review]

**Comment by cengiz-pz** [comment]
@syntaxerror247, has Github given you the option to merge the PR after I assigned the PR to you?

File: admob/addon_template/AdmobExportPlugin.gd (hunk 1 above)
Comment (syntaxerror247) [reply admob/addon_template/AdmobExportPlugin.gd]: Ok just give me few minutes

Comment (syntaxerror247) [reply admob/addon_template/AdmobExportPlugin.gd]: @cengiz-pz  what should be the error message

**Comment by syntaxerror247** [comment]
> @syntaxerror247, has Github given you the option to merge the PR after I assigned the PR to you?

No

**Comment by syntaxerror247** [comment]
@cengiz-pz error message was not added yet !

**Comment by cengiz-pz** [comment]
It's ok.  We can add it sometime before the 3.2 release.  Actually, now that I think about it, the README has to be updated as well.

**Comment by syntaxerror247** [comment]
Oh, yes

**Comment by TheSkyOne** [comment]
sorry to bump this, but will this make it possible to use the Admob script as an autoload?
currently, i cant make it an autoload script because when i do, the mentioned line `var __admob_node: Admob = _get_admob_node(EditorInterface.get_edited_scene_root())` fails to find it and causes a crash.
it would be really useful to have this be possible to use as an autoload instead of needing an instance of the node per scene.
slight issue im seeing ahead tho, is that if it was possible to make this an autoload script, all the `@export` stuff would be useless because you would have to set everything in code, which isnt the worst thing but maybe worth thinking about.
also, is there any planned date to release 3.2 where this will hopefully be solved?
i appreciate all ya'lls work!

**Comment by syntaxerror247** [comment]
@TheSkyOne Sadly this is still not possible. But you can try my method to overcome that problem.

1.Create a new scene `admanager`. Add a script, in this script create a variable `admob_node`.

2. Now just create admob node in this scene and link it to `admob_node` variable.

3. Connect all the signal and keep all your admob related logic in this script.

3. Finally Go to project settings and set this scene as an autoload.

Now you can use this in any scene, like for example `admanager.admob_node.show_banner()` and you can even create your functions here to handle all logic related to specific conditions and access like this `admanager.show_leaderboard_ad()`

**Comment by TheSkyOne** [comment]
@syntaxerror247 damn, its unfortunate that this script cant just be used as an autoload as is and needs to be wrapped in another scene...
I appreciated your guidance, and i'll be trying your suggestion soon, thank you!

**Comment by TheSkyOne** [comment]
is there any hack we can do in the meantime to not have to have the admob node in the current edited scene? it is quite frustrating tbh...

**Pull request merged**